avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -1765,6 +1782,15 @@ private CyclicSchemaWrapper cyclicSchemaWrapper(
   }
 
   private Object defaultValueFromAvro(Schema schema,
+      org.apache.avro.Schema avroSchema,
+      Object value,
+      ToConnectContext toConnectContext) {
+    Object result = defaultValueFromAvroWithoutLogical(schema, avroSchema, value, toConnectContext);
+    // If the schema is a logical type, convert the primitive Avro default into the logical form
+    return toLogical(schema, result);
+  }
```

Comment:
The old `defaultValueFromAvro(...)` method is recursive, which makes it difficult to easily convert to the logical form. These changes make it pretty easy to do that.